docs/writing-rules/pattern-syntax.mdx

```diff
@@ -515,7 +515,37 @@ When referencing an ellipsis metavariable in a rule message or [metavariable-pat
 - message: Call to foo($...ARGS)
 ```
 
-### Displaying matched metavariables in rule messages
+### Anonymous metavariables
+
+Anonymous metavariables are used to specify that a metavariable exists in the pattern you want to capture.
+
+An anonymous metavariable always takes the form `$_`. Variables such as `$_1` or `$_2` are **not** anonymous. You can use more than one anonymous metavariable in a rule definition.
+
+For example, if you want to specify that a function should **always** have 3 arguments, then you can use anonymous metavariables:
+
+```yaml
+- pattern: def function($_, $_, $_)
+```
+
+An anonymous metavariable does not produce any binding to the code it matched. This means it does not store the code that it matched. The pattern:
+
+```yaml
+- pattern: def function($A, $B, $C)
+```
+
+is not equivalent to the former example, as `$A`, `$B`, and `$C` bind to the code that matched the pattern. You can then use `$A` or any other metavariable in your rule definition to specify that specific code. Anonymous metavariables cannot be used this way.
+
+Anonymous metavariables also communicate to the reader that their values are not relevant, but rather their occurrence in the pattern.
+
+### Metavariable unification
+
+For search mode rules, metavariables with the same name are treated as the same metavariable across and within pattern operators **except for `pattern-either`**. This is called metavariable unification.
```

Comment:
hm, i might also say that it's more that this only happens across patterns combined with `patterns`

of the six fundamental pattern operators, we have `pattern`, `pattern-not`, `pattern-inside`, `pattern-regex`, `pattern-either`, and `patterns`. the first four only take in, at maximum, one pattern as an argument, so they're out. of the latter two, only `pattern-either` and `patterns` can ingest numerous patterns, and only `patterns` does unification.

i would sooner view it as `patterns` being the weird one, than `pattern-either`, personally

Reply:
Thanks for the catch!